### PR TITLE
2.0 add the invoice endpoint

### PR DESCRIPTION
**Pull request by Smitsel**

### CODE REVIEW

File: examples/20-list-invoices.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    $invoices = $mollie->invoices->all();
+    foreach ($invoices as $invoice) {
+        echo '<li><b>Invoice ' . htmlspecialchars($invoice->reference) . ':</b> (' . htmlspecialchars($invoice->issuedAt) . ')';
+        echo '<br>Status: <b>' . $invoice->status;
+        echo '<table border="1"><tr><th>Period</th><th>Description</th><th>Count</th><th>VAT Percentage</th><th>Amount</th></tr>';
+        foreach ($invoice->lines as $line) {
+            echo '<tr>';
+            echo '<td>' . htmlspecialchars($line->period) . '</td>';
+            echo '<td>' . htmlspecialchars($line->description) . '</td>';
+            echo '<td align="right">' . htmlspecialchars($line->count) . ' x</td>';
```
Comment (shirshir) [inline examples/20-list-invoices.php]: redundant `x` in the table cell? Also in the other cells.

File: examples/20-list-invoices.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            echo '<td align="right">' . htmlspecialchars($line->vatPercentage) . ' x</td>';
+            echo '<td align="right">' . htmlspecialchars($line->amount->value . " " . $line->amount->currency) . ' x</td>';
```
Comment (shirshir) [inline examples/20-list-invoices.php]: I think that normally amounts are printed currency first, ie. `EUR 15.00`. Also below in the total.

File: src/Resources/Invoice.php [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    /**
+     * Total amount of the invoice including VAT.
+     *
+     * @var object
+     */
+    public $grossAmount;
+
+    /**
+     * Object containing the invoice lines.
+     * See https://docs.mollie.com/reference/v2/invoices-api/get-invoice for reference
```
Comment (shirshir) [inline src/Resources/Invoice.php]: Not published yet?

File: tests/Mollie/API/Endpoints/InvoiceEndpointTest.php [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+        $this->assertInstanceOf(InvoiceCollection::class, $invoices);
+
+        $documentationLink = (object)['href' => 'https://docs.mollie.com/reference/v2/invoices-api/list-invoices', 'type' => 'text/html'];
+        $this->assertEquals($documentationLink, $invoices->_links->documentation);
+
+        $selfLink = (object)['href' => 'https://api.mollie.nl/v2/invoices?limit=50', 'type' => 'application/hal+json'];
+        $this->assertEquals($selfLink, $invoices->_links->self);
+
+        $this->assertEmpty($invoices->_links->previous);
+        $this->assertEmpty($invoices->_links->previous);
```
Comment (shirshir) [inline tests/Mollie/API/Endpoints/InvoiceEndpointTest.php]: should probably be `next`

File: src/Resources/Invoice.php (hunk 3 above)
Comment (Smitsel) [reply src/Resources/Invoice.php]: Yes, will publish today.

**Comment by Smitsel** [comment]
Feedback processed, also added the missing dueAt

**Pull request merged**